cdc/redo/reader/reader_test.go

```diff
@@ -252,3 +252,515 @@ func genMetaFile(t *testing.T, dir string, meta *common.LogMeta) {
 	_, err = f.Write(data)
 	require.Nil(t, err)
 }
+
+func TestLogHeapLess(t *testing.T) {
+	tests := []struct {
+		name   string
+		h      logHeap
+		i      int
+		j      int
+		expect bool
+	}{
+		{
+			name: "Delete before Update",
```

Comment:
These data is hard to read, use schema test helper to new events is preferred.

I would suggest that test for each case independently, instead of create a `tests` and loop over each.